cardano-cli/src/Cardano/CLI/EraBased/Transaction/Run.hs

```diff
@@ -1738,6 +1776,44 @@ runTransactionCalculatePlutusScriptCostCmd
           )
         $ encodePretty scriptCostOutput
 
+buildTransactionContext
+  :: ShelleyBasedEra era
+  -> SystemStartOrGenesisFileSource
+  -> MustExtendSafeZone
+  -> File EraHistory In
+  -> File () In
```

Comment:
```suggestion
  -> File (UTxO era) In
```
if it doesn't feel like shoehorning

Reply:
I left it out because it is a bit tricky since I didn't have the era in the parameters. But I may as well bite the bullet, I added a casting function for it so that I don't have to add a new error: https://github.com/IntersectMBO/cardano-cli/pull/1079/commits/6dad8d77521b9cc5195845707ae7237d813b4db7